### PR TITLE
Run CLI.Main() in Editor mode

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
The CLI script is always going to be run in edit mode (from the Unity CLI) and I figured out how to do so while using Unity normally. 

- Add custom GUI item for running `CLI.Main` (w/ shortcut)
- Add custom GUI window for updating `CLI.ProjectPath` w/ button to run `CLI.Main`
- A couple fixes to get the script running correctly again
- Created a backup of `CAVE.unity` and added the scenes folder to the gitignore

image

### CODE REVIEW

**Comment by RobertGemmaJr** [comment]
This includes #112, I'll wait for more discussion there before merging

File: .editorconfig [hunk 1]
```diff
@@ -7,6 +7,8 @@ dotnet_analyzer_diagnostic.category-Style.severity = suggestion
 # Disable var and s_ prefix warning
 dotnet_diagnostic.IDE0008.severity = none
 dotnet_diagnostic.IDE1006.severity = none
+dotnet_diagnostic.IDE0051.severity = none
+dotnet_diagnostic.RCS1213.severity = none
```
Comment (broarr) [inline .editorconfig:11]: You tell that linter who's boss!

File: unity/CAVE/Assets/Editor/CLI.cs [hunk 2]
```diff
@@ -1,58 +1,54 @@
...
+        private static Root RootX;
+        private static GameObject Root;
+        private static GameObject XrRig;
 
-        private static Story XML;
-        private static GameObject Story;
-        private static readonly Dictionary<string, (GameObject, Object)> GameObjects = new();
+        private static Dictionary<string, (GameObject, XML.Object)> GameObjects;
 
+        [MenuItem("Custom/CLI.Main %g")]
```
Comment (broarr) [inline unity/CAVE/Assets/Editor/CLI.cs:33]: oooo fancy!

File: unity/CAVE/Assets/Editor/CLI.cs [hunk 3]
```diff
@@ -69,156 +65,152 @@ public static void Main()
...
-        private static string GetXmlPathArg()
+        private static void GetXmlPathArg()
         {
             try
             {
                 string[] args = Environment.GetCommandLineArgs();
                 for (int i = 0; i < args.Length; i++)
                 {
-                    if (args[i] == "--xmlPath") { return args[++i]; }
+                    if (args[i] == "--projectPath") { ProjectPath = args[++i]; }
```
Comment (broarr) [inline unity/CAVE/Assets/Editor/CLI.cs:75]: How do the arguments interact with the scripts being run directly through unity?

File: unity/CAVE/Assets/Editor/CLI.cs [hunk 4]
```diff
@@ -69,156 +65,152 @@ public static void Main()
...
+                Debug.LogError($"ERROR: File at {ProjectPath} not found");
+                throw;
             }
-            catch (Exception e)
+            catch
             {
-                Debug.LogError($"Error: Deserialization of file at {xmlPath} failed.");
-                Debug.LogException(e);
+                Debug.LogError($"Error: Deserialization of file at {ProjectPath} failed.");
+                throw;
```
Comment (broarr) [inline unity/CAVE/Assets/Editor/CLI.cs:102]: Likewise, how are exceptions handled by Unity?

File: unity/CAVE/Assets/Editor/CLIEditor.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            EditorWindow window = GetWindow(typeof(CLI_GUI));
+            window.Show();
+        }
+
+        private void OnGUI()
+        {
+            if (CLI.ProjectPath is null) { CLI.ProjectPath = "../../test/sample.xml"; }
+
+            CLI.ProjectPath = EditorGUILayout.TextField("Path: ", CLI.ProjectPath);
```
Comment (broarr) [inline unity/CAVE/Assets/Editor/CLIEditor.cs:20]: Ahh I see, you have a full page with text fields!

File: unity/CAVE/Assets/Resources/CAVE (backup).unity [hunk 6]
```diff
@@ -0,0 +1,1445 @@
+%YAML 1.1
```
Comment (broarr) [inline unity/CAVE/Assets/Resources/CAVE (backup).unity:1]: Git is your backup! 🙂

File: unity/CAVE/Assets/Resources/CAVE.unity [hunk 7]
```diff
@@ -155,6 +155,7 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: ab68ce6587aab0146b8dabefbd806791, type: 3}
```
Comment (broarr) [inline unity/CAVE/Assets/Resources/CAVE.unity:155]: Git is your backup! 🙂

**Review by broarr — APPROVED** [review]
This looks great! Probably simplifies your workflow a lot. Is the idea that humans creating editing existing Writing 3D projects would just use the Unity editor for everything (after generating the project using that python script)?

File: unity/CAVE/Assets/Editor/CLI.cs (hunk 2 above)
Comment (RobertGemmaJr) [reply unity/CAVE/Assets/Editor/CLI.cs:33]: Keyboard shortcuts and everything!

File: unity/CAVE/Assets/Editor/CLI.cs (hunk 4 above)
Comment (RobertGemmaJr) [reply unity/CAVE/Assets/Editor/CLI.cs:102]: Unity has it's own sudo console panel where `Debug.Log`, `Debug.Error`, `Debug.Warning`, etc all show up. If it's an actual exception then execution with end and the message is printed as an error

File: unity/CAVE/Assets/Editor/CLI.cs (hunk 3 above)
Comment (RobertGemmaJr) [reply unity/CAVE/Assets/Editor/CLI.cs:75]: It's a function of `System.Environment` not Unity unfortunately. They're only passed when I call the unity editor from the command line so I have no way seeing/using them from the GUI

File: unity/CAVE/Assets/Resources/CAVE (backup).unity (hunk 6 above)
Comment (RobertGemmaJr) [reply unity/CAVE/Assets/Resources/CAVE (backup).unity]: I'll delete

File: unity/CAVE/Assets/Editor/CLI.cs (hunk 3 above)
Comment (broarr) [reply unity/CAVE/Assets/Editor/CLI.cs]: I thought that might be the case. Should we consider adding them to the GUI, and having the GUI values overwrite the environment values? Or maybe the GUI values can be prepopulated with the environment variables?

Comment (RobertGemmaJr) [reply unity/CAVE/Assets/Editor/CLI.cs]: Yes, right now the GUI and environment are both interacting with the `CLI.ProjectPath` string. The GUI checks to see if that value is null before updating it to what I've hard coded. If someone where to run the script first the GUI would see the prepopulated value

**Pull request merged**